lxd/api_internal_recover.go

```diff
@@ -417,8 +421,8 @@ func internalRecoverScan(s *state.State, userPools []api.StoragePoolsPost, valid
 
 			// Recover unknown instance volumes.
 			for _, poolVol := range poolVols {
-				if poolVol.Container == nil && poolVol.Volume != nil {
-					continue // Skip custom volumes and invalid volumes.
+				if poolVol.Container == nil && poolVol.Volume != nil || poolVol.Bucket != nil {
```

Comment:
Just double checking. Is this correct or should it be?
```go
if poolVol.Container == nil && (poolVol.Volume != nil || poolVol.Bucket != nil) {
```

Reply:
If `poolVol.Container != nil && poolVol.Volume != nil` => instance volume
If `poolVol.Container == nil && poolVol.Volume != nil` => Custom volume
If `poolVol.Bucket != nil` => Bucket volume

So this condition is meant as:
```go
if (poolVol.Container == nil && poolVol.Volume != nil) || poolVol.Bucket != nil {
//  ^ Is custom volume                                    ^ Is bucket
```

In this specific case, even `poolVol.Container == nil && (poolVol.Volume != nil || poolVol.Bucket != nil)` would work, since `poolVol.Container == nil` is true for both custom volumes, and buckets.

Which means that for clarity, I would prefer your approach.